File: src/main/java/com/hrms/hr_payroll_management_system/service/PermissionService.java
package com.hrms.hr_payroll_management_system.service;

import com.hrms.hr_payroll_management_system.dto.request.permission.CreatePermissionRequest;
import com.hrms.hr_payroll_management_system.dto.request.permission.UpdatePermissionRequest;
import com.hrms.hr_payroll_management_system.dto.response.permission.PermissionResponse;

import java.util.List;

public interface PermissionService {

    PermissionResponse create(CreatePermissionRequest request);

    List<PermissionResponse> getAll();

    PermissionResponse getById(Long id);

    PermissionResponse update(Long id, UpdatePermissionRequest request);

    void delete(Long id);

}

File: frontend/src/services/permission.service.js
import { permissionApi } from '../api/permission.api';

export const permissionService = {
  async list() {
    const res = await permissionApi.getAll();
    return res.data || [];
  },

  async create({ name, description }) {
    const res = await permissionApi.create({ name, description });
    return res.data;
  },

  async update(id, { name, description }) {
    const res = await permissionApi.update(id, { name, description });
    return res.data;
  },

  async remove(id) {
    await permissionApi.remove(id);
  },
};